### ISSUE TITLE
show welcome message only once, not on every reconnect

### ISSUE THREAD

**Issue by GoogleCodeExporter** [baluragala/gtalksms#95]
```
What steps will reproduce the problem?
1. switch on phone's wifi
2. turn it to be active only on screen on or charger connected
3. start gtalksms up
4. let the screen turn off.
5. in a couple of minutes the phone will disconnect from WiFi and switch to GSM 
network, causing gtalksms reconnect

What version of the product are you using? On what operating system?
Gtalksms 1.9, Android 2.2.2, on NexusOne


Please provide any additional information below.
If you are using the phone heavily(often switch it on, and let it turn the 
screen off after) it would send you welcome message every time it reconnects: 
one time it switches to WiFi, one time it switches to GSM, that pretty annoys 
filling your XMPP log w/ messages like this:
My Phone: Welcome to GTalkSMS v1.9 @ Yakoo. Send "?" for getting help.

Wouldn't it be pretty if it was displayed only on first connection since 
application has started?
```

Original issue reported on code.google.com by `me.detected` on 10 Feb 2011 at 3:50

**Comment by GoogleCodeExporter** [comment]
```
Yes, this behavior is annoying. Until we have fixed this, disable the welcome 
message in the settings.

This fix wouldn't be so trivial, as android has the ability to kill and restart 
the service in order to reduce his heap size when its under pressure. We have 
to add an state, remember this on persistent storage and check the state on 
restart.
```

Original comment by `fschm...@gmail.com` on 10 Feb 2011 at 6:15
- Changed state: **Accepted**
- Added labels: **Priority-Low**
- Removed labels: **Priority-Medium**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 10 Feb 2011 at 6:22
- Changed title: **show welcome message only once, not on every reconnect**
- Changed state: **New**

**Comment by GoogleCodeExporter** [comment]
Original comment by `fschm...@gmail.com` on 10 Feb 2011 at 9:42
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
This seems more trouble than it is worth to me - why have a "welcome" message 
at all?  The fact the app "started" is just an implementation detail - and as 
Florent said, an implementation detail to the *device* that we don't fully 
control.  What you care about is when it is actually connected as that is the 
point of the app.  Just rename it to a "connection message" and be done :)

To put it another way - what is lost for people when they disable this message? 
 I'd suggest nothing, as the xmpp protocol already indicates if it is connected 
or not.

```

Original comment by `skippy.hammond@gmail.com` on 10 Feb 2011 at 11:57

**Comment by GoogleCodeExporter** [comment]
```
I also always wondered why we have this welcome message. I have no problem with 
renaming the function or even droping the code at all. In fact, this would be 
my prefered option. 
```

Original comment by `fschm...@gmail.com` on 11 Feb 2011 at 6:42

**Comment by GoogleCodeExporter** [comment]
```
This "welcome" message is mostly a message issued on successful connection of 
gtalksms. That's why the preference says: "Application connection" and "Phone 
reconnect notification".
Sending it only once would give the user no or just a marginal benefit. So the 
best strategy is to disable this message at all. :)
```

Original comment by `fschm...@gmail.com` on 10 Mar 2011 at 9:55
- Changed state: **WontFix**

**Comment by GoogleCodeExporter** [comment]
```
This seems to be forgotten... the "disable welcome message" was a good 
compromise, however it seems to have vanished from recent versions ?
```

Original comment by `neam...@gmail.com` on 26 Sep 2011 at 7:42

**Comment by GoogleCodeExporter** [comment]
```
The welcome message was removed completely from GTalkSMS, as it provides no 
benefit, because you will see the contact coming online when GTalkSMS 
successfully connects.
```

Original comment by `fschm...@gmail.com` on 27 Sep 2011 at 9:09

**Comment by GoogleCodeExporter** [comment]
```
That’s great, however i'm using the last version from the market (3. 2)  and 
i still get this annoying notification.
```

Original comment by `neam...@gmail.com` on 27 Sep 2011 at 9:16

**Comment by GoogleCodeExporter** [comment]
```
Ups sorry, the code was not removed. You can disable it under preferences -> 
notification settings -> application connection
```

Original comment by `fschm...@gmail.com` on 27 Sep 2011 at 9:27
